### ISSUE TITLE
Document side-effects of increasing the number of threads

### ISSUE THREAD

**Issue by danieldk** [finalfusion/finalfrontier#74]

**Comment by danieldk** [comment]
I think that after #89 this is the only item left. Where do we add this? To the finalfrontier man pages? Perhaps also to the quickstart?

**Comment by sebpuetz** [comment]
~~~Rust
if n_threads > 18 {
    eprintln!("You're treading on thin ice.")
}
~~~

or more seriously:

~~~Rust
if n_threads > 18 {
    eprintln!("Training with a large number of threads can lead to bad convergence, see: {}", url_to_explanation);
}
~~~

Otherwise, note something in `n_threads`' `finalfrontier --help` entry with a link to the page with explanations. That's probably the go-to for most people. Since we keep the manpage in sync with the cmd args, the note on `n_threads` from `--help` could also go there.

 I'd say that page should be a) inside this repo and b) on the finalfrontier page on the website.

**Comment by danieldk** [comment]
18 would be a bit strange, since we cap at 20 threads by default, so any machine >36 virtual CPUs would spit out this warning by default ;). I am not sure if we should make the command-line tools annoy people who actually want to do this (and simply train for more epochs). What number is acceptable is also highly dependent on the data (e.g. consider tag embeddings vs. word embeddings).

As for `--help`, the layout is already terrible on 80x25 terminals as it is :(.

I think this should just go into the manuals, so man pages, quickstart, and possibly the README.

**Comment by sebpuetz** [comment]
I wrote 18 because that was a number given in the paper you linked in the other issue, no other motivation there.

> I am not sure if we should make the command-line tools annoy people who actually want to do this (and simply train for more epochs).

Not sure how annoying a line on stderr would be, but I guess the default cap 

> I think this should just go into the manuals, so man pages, quickstart, and possibly the README.

Quickstart and README should be fine then, just somewhere that people look at. I'd usually go for a) github repo or b) --help to find out about parameters, and the README would cover that scenario.

**Comment by danieldk** [comment]
> Quickstart and README should be fine then, just somewhere that people look at. I'd usually go for a) github repo or b) --help to find out about parameters, and the README would cover that scenario.

I guess I am too old ;), for me the man page or perhaps `--help` is the first place I look at. I like self-contained UNIX systems where you do not have to hunt for online documentation. Plus, man pages are trivially searchable.

**Comment by danieldk** [comment]
Fixed in d612dcd282b5ad61e6e45e8ce170498c05e53134 and 64429da01d1b6e5f8fbd8e59439cdd33abcbcdc3.
